src/Stiefel.jl

```diff
@@ -0,0 +1,239 @@
+using LinearAlgebra: diag, qr, tr, svd, mul!, zeros, lyap
+
+@doc doc"""
+    Stiefel{m,n,T} <: Manifold
+
+The Stiefel manifold consists of all $m\times n$, $m\geq n$ orthonormal matrices, i.e.
+
+````math
+\mathcal M = \{ x \in \mathbb K^{m\times n} : \overline{x}^{\mathrm{T}}x = I_n \}
```

Comment:
Is K for the field more common than F? I remember you'd said earlier that Germans use K.

Reply:
What does the overline signify? Complex conjugation? If so, perhaps the superscript H for the adjoint would be more clear.

Reply:
Oh, sorry, I meant to use F (for Field), germans still use K (which stands for Körper that translates (literally) to body but is the German mathemetical term for a field). I will change that.

Yes overline transposed is the complex conjugate – I can add that. I have used that notation for the Grassmann manifold already, but I can also change that to H. I personally prefer `$\bar{\cdot}^{\mathrm{T}}$` but if we in general switch to `$\cdot^\mathrm{H}$` I won't mind.

Reply:
I figured it was that. :)

I don't have a strong preference here (my own writing switches between 4 notations for the adjoint, that's how undecided I am). The reason I (currently) like the H superscript is because it emphasizes its connection to the transpose (in nearly all cases I've encountered, the adjoint for complex matrices is more like the transpose for real than the transpose for complex is), and it's unique; I've never used the H superscript for anything else, whereas overlines have meanings in stats, asterisks have meanings in differential geometry, and apostrophes have meanings in calculus.

🤷‍♂ But I might feel differently in a week.